packages/contracts/test/registry.ts

```diff
@@ -1,46 +1,58 @@
-import {expect} from 'chai';
-import {ethers} from 'hardhat';
-import {customError} from './test-utils/custom-error-helper';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { customError } from './test-utils/custom-error-helper';
 
-const EVENTS = {
-  NewDAORegistered: 'NewDAORegistered',
-};
+import { DAO, Registry } from '../typechain';
+
+const REGISTER_DAO_ROLE = ethers.utils.id('REGISTER_DAO_ROLE')
 
 describe('Registry', function () {
-  let registry: any;
+  let registry: Registry;
+  let dao: DAO;
   let ownerAddress: string;
 
-  before(async () => {
-    const signers = await ethers.getSigners();
+  beforeEach(async function () {
+    const signers = await ethers.getSigners()
     ownerAddress = await signers[0].getAddress();
-  });
 
-  beforeEach(async function () {
+    const DAO = await ethers.getContractFactory('DAO');
+    dao = await DAO.deploy();
```

Comment:
Isn't the DAO factory expected... to already have a Registry to register DAOs to?
How is it that we're deploying the registry afterwards?

Reply:
Can you please eloborate further?
This code piece deploys the owning DAO of the component (here for testing purposes) as explained above.

Reply:
Right, this is related to the topic above. IMO, I would try to keep things as simple as possible. The more "variables" that we put into the equation, the harder it may become to realize issues when testing ;)

Reply:
But this is nothing unusual. We also do it like that in all other component tests (`ERC20Voting`, `WhitelistVoting`, `MerkleMinter`, `MerkleDistributor`).